newrelic/resource_newrelic_alert_plugins_condition_test.go

```diff
@@ -0,0 +1,513 @@
+	client := testAccProvider.Meta().(*ProviderConfig).Client
+	for _, r := range s.RootModule().Resources {
+		ids, err := parseIDs(r.Primary.ID, 2)
+		if err != nil {
+			return err
+		}
+
+		policyID := ids[0]
+		id := ids[1]
+
+		_, err = client.GetAlertCondition(policyID, id)
+		if err == nil {
+			return fmt.Errorf("alert condition still exists")
+		}
+
+	}
+	return nil
+}
+
+func testAccCheckNewRelicAlertPluginsConditionExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no alert condition ID is set")
+		}
+
+		client := testAccProvider.Meta().(*ProviderConfig).Client
+
+		ids, err := parseIDs(rs.Primary.ID, 2)
+		if err != nil {
+			return err
+		}
+
+		policyID := ids[0]
+		id := ids[1]
+
+		found, err := client.GetAlertCondition(policyID, id)
+		if err != nil {
+			return err
+		}
+
+		if found.ID != id {
+			return fmt.Errorf("alert condition not found: %v - %v", id, found)
+		}
+
+		return nil
+	}
+}
+
+// TODO
+func TestErrorThrownUponPluginsConditionNameGreaterThan64Char(t *testing.T) {
+	expectedErrorMsg, _ := regexp.Compile(`expected length of name to be in the range \(1 \- 64\)`)
+	rName := acctest.RandString(5)
+	resource.Test(t, resource.TestCase{
+		IsUnitTest: true,
+		Providers:  testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testErrorThrownUponPluginsConditionNameGreaterThan64Char(rName),
+				ExpectError: expectedErrorMsg,
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicAlertPluginsConditionConfig(rName string) string {
+	return fmt.Sprintf(`
+data "newrelic_application" "app" {
+	name = "%[2]s"
+}
+
+resource "newrelic_alert_policy" "foo" {
+  name = "tf-test-%[1]s"
+}
+
+resource "newrelic_alert_plugins_condition" "foo" {
+  policy_id = "${newrelic_alert_policy.foo.id}"
+
+  name               = "tf-test-%[1]s"
+  enabled            = false
+  entities           = ["${data.newrelic_application.app.id}"]
+  metric             = "my-metric"
+  runbook_url        = "https://foo.example.com"
+  metric_description = "my-metric-description"
+  plugin_id          = "123"
+  plugin_guid        = "com.example.plugin"
+  value_function     = "average"
+  condition_scope    = "application"
+
+  term {
+    duration      = 5
+    operator      = "below"
+    priority      = "critical"
+    threshold     = "0.75"
+    time_function = "all"
+  }
+}
+`, rName, testAccExpectedApplicationName)
+}
+
+func testAccCheckNewRelicAlertPluginsConditionConfigUpdated(rName string) string {
+	return fmt.Sprintf(`
+data "newrelic_application" "app" {
+	name = "%[2]s"
+}
+
+resource "newrelic_alert_policy" "foo" {
+  name = "tf-test-updated-%[1]s"
+}
+
+resource "newrelic_alert_plugins_condition" "foo" {
+  policy_id = "${newrelic_alert_policy.foo.id}"
+
+  name               = "tf-test-%[1]s"
```

Comment:
This attribute should be set to `"tf-test-updated-%[1]s"` to match the assertion above.